pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -162,6 +163,17 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
+	isCCS := accountClaim.Spec.BYOCAWSAccountID != ""
+
+	if accountClaim.Status.State == awsv1alpha1.ClaimStatusPending {
+		now := metav1.Now()
+		unclaimedCondition := controllerutils.FindAccountClaimCondition(accountClaim.Status.Conditions, awsv1alpha1.AccountUnclaimed)
+		if unclaimedCondition != nil {
+			pendingDuration := now.Sub(unclaimedCondition.LastTransitionTime.Time)
+			localmetrics.Collector.SetAccountClaimPendingDuration(isCCS, pendingDuration.Seconds())
+		}
+	}
+
```

Comment:
I don't think this is getting called for CCS accounts, see line 156 above, we're returning if it's a BYOC account.

I think we should abstract this to a function and utilize it both here on this line as well as around line 410 where we're requeueing while waiting.